### PR TITLE
ElmerAdrianV/mvp/profile user view

### PR DESCRIPTION
**Pull request by ElmerAdrianV**

### CODE REVIEW

File: app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+        view.findViewById(R.id.btnSaveChanges).setOnClickListener(v -> setupSaveChanges(editUserFields));
+        view.findViewById(R.id.btnLogout).setOnClickListener(v -> setupLogout());
+        view.findViewById(R.id.btnAddContact).setOnClickListener(v -> setupAddContact(view));
+        view.findViewById(R.id.btnAdd).setOnClickListener(v -> setupAdd(view));
+    }
+
+    private void setupAdd(View view) {
+        String nickname=((EditText)view.findViewById(R.id.etNickname)).getText().toString();
+        String number=((EditText) view.findViewById(R.id.etPhoneNumber)).getText().toString();
+        EmergencyContacts contact = new EmergencyContacts();
```
Comment (jmccormack200) [inline app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java:93]: Let's add a TODO here to move to a "parse sublcass". Let's do the work in another PR but we can use the docs here to make this a bit easier to work with:

https://docs.parseplatform.org/android/guide/#subclasses

File: app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        if(nickname.isEmpty() && number.isEmpty()){
+            Toast.makeText(getContext(),"Please, fill all the fields",Toast.LENGTH_SHORT).show();
+        }
+        else{
+            contact.put(EmergencyContacts.USER_KEY,currentUser);
+            contact.put(EmergencyContacts.NICKNAME_KEY,nickname);
+            contact.put(EmergencyContacts.NUMBER_KEY,new Long(number));
+            contact.saveInBackground(e -> {
+                if(e!=null){
```
Comment (jmccormack200) [inline app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java:102]: is this the right logic here? What happens if we get an error here?

File: app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    private void putProfileImage() {
+        int radius = 100000;
+        Glide.with(getContext()).load(profilePhoto.getUrl())
+                .fitCenter() // scale image to fill the entire ImageView
+                .transform(new RoundedCorners(radius))
+                .into(ivProfileView);
+    }
+
+    private void setupSaveChanges(HashMap<String, MyPair<EditText, Boolean>> editUserFields) {
+        MyPair<EditText, Boolean> pair;
```
Comment (jmccormack200) [inline app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java:140]: do we need to set this up before the for loop?

File: app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        for (String key : editUserFields.keySet()) {
+            pair = editUserFields.get(key);
+            if (pair.second) {
```
Comment (jmccormack200) [inline app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java:143]: do we actually need "pair.second"? could we just check if the edit text is empty or not?

File: app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+                pair.second = false;
+                pair.first.setEnabled(false);
+            }
+        }
+    }
+
+    private void setupRecycleView(View view) {
+        RecyclerView rvReportCard = view.findViewById(R.id.rvEmergencyContacts);
```
Comment (jmccormack200) [inline app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java:174]: nice, this is a great setup function

File: app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+import com.parse.ParseFile;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+
+import java.io.File;
+import java.util.HashMap;
+
+
+public class ProfileUserFragment extends Fragment {
+    protected EmergencyContactsCardAdapter adapter;
```
Comment (jmccormack200) [inline app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java:44]: Let's take a look at all these member variables. What ones actually need to be member variables and what ones could be local variables?

File: app/src/main/res/layout/fragment_profile_user.xml [hunk 7]
```diff
@@ -0,0 +1,394 @@
...
+    <ScrollView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <RelativeLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent">
+
+
+            <ImageButton
```
Comment (jmccormack200) [inline app/src/main/res/layout/fragment_profile_user.xml:24]: make sure to run the formatter here

File: app/src/main/res/layout/fragment_profile_user.xml [hunk 8]
```diff
@@ -0,0 +1,394 @@
...
+                android:layout_below="@+id/etEmergencyMessage"
+                android:layout_alignParentStart="true"
+                android:layout_marginStart="@dimen/separation_border_parent"
+                android:layout_marginTop="10dp"
+                android:text="@string/your_emergency_contacts"
+                android:textColor="#000000"
+                android:textSize="20sp"
+                android:textStyle="bold" />
+
+            <androidx.cardview.widget.CardView
```
Comment (jmccormack200) [inline app/src/main/res/layout/fragment_profile_user.xml:295]: We should break this cardview out into another layout file. We generally want to avoid relative layouts nested into one another.

Comment (jmccormack200) [reply app/src/main/res/layout/fragment_profile_user.xml:295]: It might make sense to move the new contact logic into a separate fragment and then navigate _back_ here after. Right now it looks like you have a few nested relative layouts + a recycler view at the end which can cause problems if a recycler view is nested inside another scroll view.

File: app/src/main/java/com/elmeradrianv/shesafe/adapters/ReportCardAdapter.java [hunk 9]
```diff
@@ -42,7 +40,7 @@ public void addAll(List<Report> list) {
     @NonNull
     @Override
     public ReportCardAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
-        View view = LayoutInflater.from(context).inflate(R.layout.report_card_view, parent, false);
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.report_card_view, parent, false);
```
[the remark this answers is not in the document]
Comment (ElmerAdrianV) [reply app/src/main/java/com/elmeradrianv/shesafe/adapters/ReportCardAdapter.java:42]: Yeah, I resolve that conversation in this pull request for avoid merge conflicts

File: app/src/main/res/layout/fragment_profile_user.xml (hunk 8 above)
Comment (jmccormack200) [reply app/src/main/res/layout/fragment_profile_user.xml:296]: Note: We'll do this in a TODO at a later time.

**Review by jmccormack200 — APPROVED** [review]
